src/app/problem/editor/right/ProvidedCodes.js

```diff
@@ -0,0 +1,95 @@
+import Block from "./Block";
+import {useEffect, useState} from "react";
+import {EditSaveCancelButton} from "../../commons/EditSaveCancelButton";
+import {useUploads} from "../../../usecases/UploadFilesUseCase";
+import {ACTION_UPDATE_PROVIDEDCODES, useProblemEditorContext} from "../context";
+import IconTextItems from "../../../commons/TextInputForm/IconTextItems";
+import FixedUploadFileItems from "./FixedUploadFileItems";
+import UploadFileButton from "../../commons/UploadFileButton";
+import {VscFileCode} from "react-icons/all";
+import {problemService} from "../../../../services/services";
+
+
+const ProvidedCodes = () => {
+    const {problem, dispatch} = useProblemEditorContext();
+    const [isEditing, setIsEditing] = useState(false);
+    const [providedCodesFileId, setProvidedCodesFileId] = useState(undefined);
+    const [providedCodesFileNames, setProvidedCodesFileNames] = useState([]);
+    const {files, setFiles, addFiles, removeFile} = useUploads();
+
+    useEffect(() => {
+        if (problem) {
+            const providedCodes = problem.languageEnvs[0].providedCodes;
+            if (!providedCodesFileId && providedCodes) {
+                setProvidedCodesFileNames(providedCodes.providedCodesFileNames);
+                setProvidedCodesFileId(providedCodes.providedCodesFileId);
+            }
+        }
+    }, [problem, providedCodesFileId, setProvidedCodesFileId, setProvidedCodesFileNames]);
```

Comment:
Don't split it into two states, do the following instead.
 
`const [providedCodes, setProvidedCodes] = useState(undefined);`

so you may write something like below in `useEffect`

```js
useEffect(() => {
        if (problem) {
            if (!providedCodes &&  problem.languageEnvs[0].providedCodes) {
                setProvidedCodes(problem.languageEnvs[0].providedCodes);
            }
        }
    }, [problem, providedCodes, setProvidedCodes]);
```